File: rater/main/forecastRatings.m
function [tTree mTree] = forecastRatings(currentDate)
  setup;
  [tTree fTree mTree T hA] = readData(currentDate, dataPath);
  mi = MatchIterator(mTree);
  [mTree mi] = normalizeMatchGoals(mTree, mi, hA);
  [tTree mTree mi] = optimizeRatings(tTree, fTree, mTree, mi, ...
      height(T));
  vTree = verifyModel(mi);
  displayResults(vTree);
  T = writeData(mi, T, dataPath);
end

File: rater/main/forecastRatings.R
forecastRatings <- function(currentDate, currentContest, rData=NULL) {
  library(hash)
  library(parallel)
  library(MASS)
  srcFiles <- list.files("../", ".*\\.R",
      full.names=TRUE, recursive=TRUE)
  sapply(srcFiles, source)
  dateFormat <- "%m/%d/%y"
  dataPath <- "../../data/"
  currentDate <- as.Date(currentDate, dateFormat)
  readsData <- readData(currentDate, currentContest,
      dateFormat, dataPath)
  tTree <- readsData[["tTree"]]
  fTree <- readsData[["fTree"]]
  gTree <- readsData[["gTree"]]
  T <- readsData[["T"]]
  gi <- newGameIterator(gTree)
  optPrereqs <- computeOptPrereqs(gTree, gi, T)
  rData <- optimizeRatings(tTree, fTree, optPrereqs, relevantGoals,
      rData)
  writeData(rData, T, dataPath)
  rData
}
